src/rules/noUnnecessaryCastRule.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.TypedRule {
+  /* tslint:disable:object-literal-sort-keys */
+  public static metadata: Lint.IRuleMetadata = {
+    ruleName: "no-unnecessary-cast",
+    description: `Warns if a cast does not change the type of an expression.`,
+    options: null,
+    optionsDescription: "Not configurable",
+    type: "typescript",
+    typescriptOnly: true,
+    requiresTypeInfo: true,
+  };
+  /* tslint:enable:object-literal-sort-keys */
+
+  public static FAILURE_STRING = "This cast is unnecessary.";
```

Comment:
Added suggested failure string and renamed the rule everywhere. The AST is a little misleading here since it refers to "<type>expr" as a type assertion and the others as something else.